<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genomic annotation and publication-grade visualization • GAnnoViz</title>
<!-- katex math --><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.css" integrity="sha384-nB0miv6/jRmo5UMMR1wu3Gz6NLsoTkbqJghGIsx//Rlm+ZU03BU6SQNC66uf4l5+" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.js" integrity="sha384-7zkQWkzuo3B5mTepMUcHkMB5jZaolc2xDwL6VFqjFALcbeS9Ggm/Yr2r3Dy4lfFg" crossorigin="anonymous"></script><script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/contrib/auto-render.min.js" integrity="sha384-43gviWU0YVjaDtb/GhzOouOXtZMP/7XUzwPTstBeZFe/+rCMvRwr4yROQP43s0Xk" crossorigin="anonymous" onload="renderMathInElement(document.body);"></script><script src="katex-auto.js"></script><script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Genomic annotation and publication-grade visualization">
<meta name="description" content="Genome annotation and visualization for exploratory and publication use. Supports gene/interval/chromosome structures, genome-wide density maps, circos-style rings, and signal overlays. Provides robust window aggregation for FST and DMR, and consistent theming for reproducible figures.">
<meta property="og:description" content="Genome annotation and visualization for exploratory and publication use. Supports gene/interval/chromosome structures, genome-wide density maps, circos-style rings, and signal overlays. Provides robust window aggregation for FST and DMR, and consistent theming for reproducible figures.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">GAnnoViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html"><span class="fa fa-passport"></span> Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-file-code"></span> Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/Tutorial.html"><span class="fa fa-file-code"></span> Tutorial</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/benben-miao/GAnnoViz/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gannoviz">GAnnoViz<a class="anchor" aria-label="anchor" href="#gannoviz"></a>
</h1></div>
<ul>
<li>
<a href="#gannoviz">GAnnoViz</a>
<ul>
<li><a href="#id_1-introduction">1. Introduction</a></li>
<li><a href="#id_2-installation">2. Installation</a></li>
<li><a href="#id_3-shiny-app">3. Shiny App</a></li>
<li><a href="#id_4-extract-features">4. Extract Features</a></li>
<li><a href="#id_5-plot-structure">5. Plot Structure</a></li>
<li><a href="#id_6-deg-anno--viz">6. DEG Anno &amp; Viz</a></li>
<li><a href="#id_7-snp-anno--plot">7. SNP Anno &amp; Plot</a></li>
<li><a href="#id_8-dmg-anno--plot">8. DMG Anno &amp; Plot</a></li>
</ul>
</li>
</ul>
</div>
<div class="section level1">
<h1 id="gannoviz-1">GAnnoViz<a class="anchor" aria-label="anchor" href="#gannoviz-1"></a>
</h1>
<div class="section level2">
<h2 id="id_1-introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#id_1-introduction"></a>
</h2>
<p><strong>GAnnoViz: a R package for genomic annotation and visualization.</strong></p>
<p><strong>GAnnoViz (genomic annotation and visualization)</strong> is an R ecosystem specialized for annotating and visualizing chromosome-level gene functional elements in multi-omics. It provides core analysis and visualization functions through an R package and drives an online platform to enable experimental data analysis through real-time interaction. Currently, GAnnoViz includes 26 functions and 3 datasets. These functions are mainly used to address analytical needs in the following four aspects: extraction of genomic functional elements, drawing of linear gene structures, multi-omics sliding window annotation, and visualization of chromosome-level gene expression.</p>
<p>These functions are compatible with each other in terms of input and output results, collaborating freely and closely to complete complex analytical tasks. Although this project uses the genome of the model species mouse as a reference, it has been tested that standard structural annotations and omics data from any species are supported. It is worth noting that the results of transcriptome differential expression should be in the DESeq2 structure, while epigenome differential methylation data should refer to the MethylKit results as the standard.</p>
<p>GAnnoViz prioritizes the use of <code>GRanges</code> objects for parsing gene structures and interval annotations, and uses <code>ggplot</code> objects as the standard for graph drawing, facilitating high customization and long-term maintenance. This ecosystem adopts an open-source approach for suggestion adoption, new function iteration, continuous community maintenance, and is supported by the Hiplot cloud platform. The complete local or online help documentation contains detailed parameter descriptions and runnable examples of all functions, as well as a streamlined tutorial protocol.</p>
<p><strong>Open Source Code:</strong> <a href="https://github.com/benben-miao/GAnnoViz/" class="uri">https://github.com/benben-miao/GAnnoViz/</a></p>
<p><strong>Documents API</strong>: <a href="https://benben-miao.github.io/GAnnoViz/" class="external-link uri">https://benben-miao.github.io/GAnnoViz/</a></p>
<p><strong>Cloud Platform</strong>: <a href="https://shiny.hiplot.cn/gannoviz/" class="external-link uri">https://shiny.hiplot.cn/gannoviz/</a></p>
<figure><img src="https://benben-miao.github.io/GAnnoViz/image/Shinyapp.png" alt="Shinyapp UI"><figcaption aria-hidden="true">
Shinyapp UI
</figcaption></figure>
</div>
<div class="section level2">
<h2 id="id_2-installation">2. Installation<a class="anchor" aria-label="anchor" href="#id_2-installation"></a>
</h2>
</div>
<div class="section level2">
<h2 id="id_3-shiny-app">3. Shiny App<a class="anchor" aria-label="anchor" href="#id_3-shiny-app"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GAnnoVizApp()</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="id_4-extract-features">4. Extract Features<a class="anchor" aria-label="anchor" href="#id_4-extract-features"></a>
</h2>
<div class="section level3">
<h3 id="extract-genes">Extract Genes<a class="anchor" aria-label="anchor" href="#extract-genes"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example GFF3 file in GAnnoViz</span></span>
<span><span class="va">gff_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>,</span>
<span>  <span class="st">"example.gff3.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract Genes</span></span>
<span><span class="co"># genes &lt;- extract_genes(</span></span>
<span><span class="co">#   gff_file = gff_file,</span></span>
<span><span class="co">#   format = "auto",</span></span>
<span><span class="co">#   gene_info = "all")</span></span>
<span><span class="co"># genes</span></span>
<span></span>
<span><span class="co"># Gene info: gene_range</span></span>
<span><span class="va">gene_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_genes.html">extract_genes</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  gene_info <span class="op">=</span> <span class="st">"gene_range"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_range</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 6 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;                          start       end     width</span></span>
<span><span class="co">#&gt;                      &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000001 108014596 108053462     38867</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000003  76881507  76897229     15723</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000028  18599197  18630737     31541</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000037 159865521 160041209    175689</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000049 108234180 108305222     71043</span></span>
<span><span class="co">#&gt;   ENSMUSG00000000056 121128079 121146682     18604</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-cds">Extract CDS<a class="anchor" aria-label="anchor" href="#extract-cds"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract CDS</span></span>
<span><span class="co"># cds &lt;- extract_cds(</span></span>
<span><span class="co">#   gff_file = gff_file,</span></span>
<span><span class="co">#   format = "auto",</span></span>
<span><span class="co">#   cds_info = "all")</span></span>
<span><span class="co"># cds</span></span>
<span></span>
<span><span class="co"># CDS info: cds_range</span></span>
<span><span class="va">cds_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_cds.html">extract_cds</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  cds_info <span class="op">=</span> <span class="st">"cds_range"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cds_range</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 0 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;        start       end     width</span></span>
<span><span class="co">#&gt;    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-promoters">Extract Promoters<a class="anchor" aria-label="anchor" href="#extract-promoters"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract Promoters</span></span>
<span><span class="co"># promoters &lt;- extract_promoters(</span></span>
<span><span class="co">#   gff_file = gff_file,</span></span>
<span><span class="co">#   format = "auto",</span></span>
<span><span class="co">#   upstream = 2000,</span></span>
<span><span class="co">#   downstream = 200,</span></span>
<span><span class="co">#   promoter_info = "all")</span></span>
<span><span class="co"># promoters</span></span>
<span></span>
<span><span class="co"># Promoter info: promoter_range</span></span>
<span><span class="va">promoter_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_promoters.html">extract_promoters</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  promoter_info <span class="op">=</span> <span class="st">"promoter_range"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">promoter_range</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 6 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;                   start       end     width</span></span>
<span><span class="co">#&gt;               &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;    Lypla1-205   4876011   4878210      2200</span></span>
<span><span class="co">#&gt;    Lypla1-201   4876046   4878245      2200</span></span>
<span><span class="co">#&gt;    Lypla1-208   4876053   4878252      2200</span></span>
<span><span class="co">#&gt;   Gm37988-201   4876115   4878314      2200</span></span>
<span><span class="co">#&gt;    Lypla1-203   4876119   4878318      2200</span></span>
<span><span class="co">#&gt;    Lypla1-206   4876121   4878320      2200</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-5utr">Extract 5’UTR<a class="anchor" aria-label="anchor" href="#extract-5utr"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract 5'UTR</span></span>
<span><span class="co"># utr5 &lt;- extract_utr5(</span></span>
<span><span class="co">#   gff_file = gff_file,</span></span>
<span><span class="co">#   format = "auto",</span></span>
<span><span class="co">#   utr5_info = "all")</span></span>
<span><span class="co"># utr5</span></span>
<span></span>
<span><span class="co"># 5'UTR info: utr5_range</span></span>
<span><span class="va">utr5_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/extract_utr5.html">extract_utr5</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  utr5_info <span class="op">=</span> <span class="st">"utr5_range"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">utr5_range</span><span class="op">)</span></span>
<span><span class="co">#&gt; IRanges object with 0 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;        start       end     width</span></span>
<span><span class="co">#&gt;    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="id_5-plot-structure">5. Plot Structure<a class="anchor" aria-label="anchor" href="#id_5-plot-structure"></a>
</h2>
<div class="section level3">
<h3 id="plot-gene-stats-for-chromosomes">Plot gene stats for chromosomes<a class="anchor" aria-label="anchor" href="#plot-gene-stats-for-chromosomes"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot gene stats</span></span>
<span><span class="fu"><a href="reference/plot_gene_stats.html">plot_gene_stats</a></span><span class="op">(</span></span>
<span>    gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>    format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>    bar_width <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    bar_color <span class="op">=</span> <span class="st">"#0055ff55"</span>,</span>
<span>    lable_size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_gene_stats-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-gene-structure-promoter-3utr-exon-intron-5utr">Plot gene structure (Promoter, 3’UTR, Exon, Intron, 5’UTR)<a class="anchor" aria-label="anchor" href="#plot-gene-structure-promoter-3utr-exon-intron-5utr"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot gene structure</span></span>
<span><span class="fu"><a href="reference/plot_gene_structure.html">plot_gene_structure</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSMUSG00000025935"</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  feature_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  intron_width <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  x_breaks <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  arrow_length <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  arrow_count <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  arrow_unit <span class="op">=</span> <span class="st">"pt"</span>,</span>
<span>  promoter_color <span class="op">=</span> <span class="st">"#ff8800"</span>,</span>
<span>  utr5_color <span class="op">=</span> <span class="st">"#008833"</span>,</span>
<span>  utr3_color <span class="op">=</span> <span class="st">"#ff0033"</span>,</span>
<span>  exon_color <span class="op">=</span> <span class="st">"#0033ff"</span>,</span>
<span>  intron_color <span class="op">=</span> <span class="st">"#333333"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_gene_structure-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-protein-domains-from-ensembl">Plot protein domains from Ensembl<a class="anchor" aria-label="anchor" href="#plot-protein-domains-from-ensembl"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot TP53 domian</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plot_gene_domains.html">plot_gene_domains</a></span><span class="op">(</span></span>
<span>  gene_name <span class="op">=</span> <span class="st">"TP53"</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"hsapiens"</span>,</span>
<span>  transcript_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  transcript_choice <span class="op">=</span> <span class="st">"longest"</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"Set 2"</span>,</span>
<span>  legend_ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  return_data <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_gene_domains-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-gene-structures-for-a-genomic-interval">Plot gene structures for a genomic interval<a class="anchor" aria-label="anchor" href="#plot-gene-structures-for-a-genomic-interval"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot interval structure</span></span>
<span><span class="fu"><a href="reference/plot_interval_structure.html">plot_interval_structure</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  chrom_id <span class="op">=</span> <span class="st">"chr1"</span>,</span>
<span>  start <span class="op">=</span> <span class="fl">13600000</span>,</span>
<span>  end <span class="op">=</span> <span class="fl">13800000</span>,</span>
<span>  x_breaks <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  feature_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  intron_width <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  arrow_count <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  arrow_length <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  arrow_unit <span class="op">=</span> <span class="st">"pt"</span>,</span>
<span>  promoter_color <span class="op">=</span> <span class="st">"#ff8800"</span>,</span>
<span>  utr5_color <span class="op">=</span> <span class="st">"#008833"</span>,</span>
<span>  utr3_color <span class="op">=</span> <span class="st">"#ff0033"</span>,</span>
<span>  exon_color <span class="op">=</span> <span class="st">"#0033ff"</span>,</span>
<span>  intron_color <span class="op">=</span> <span class="st">"#333333"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_interval_structure-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-interval-flank-structure-around-a-focal-gene">Plot interval flank structure around a focal gene<a class="anchor" aria-label="anchor" href="#plot-interval-flank-structure-around-a-focal-gene"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Neighborhood around a focal gene on its chromosome</span></span>
<span><span class="fu"><a href="reference/plot_interval_flank.html">plot_interval_flank</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  gene_id <span class="op">=</span> <span class="st">"ENSMUSG00000025935"</span>,</span>
<span>  flank_upstream <span class="op">=</span> <span class="fl">200000</span>,</span>
<span>  flank_downstream <span class="op">=</span> <span class="fl">200000</span>,</span>
<span>  show_promoters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  arrow_length <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  arrow_unit <span class="op">=</span> <span class="st">"pt"</span>,</span>
<span>  gene_color <span class="op">=</span> <span class="st">"#0088ff"</span>,</span>
<span>  promoter_color <span class="op">=</span> <span class="st">"#ff8800"</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_interval_flank-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-chromosome-structures-and-gene-stats">Plot chromosome structures and gene stats<a class="anchor" aria-label="anchor" href="#plot-chromosome-structures-and-gene-stats"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot chrom structure</span></span>
<span><span class="fu"><a href="reference/plot_chrom_structure.html">plot_chrom_structure</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  bar_width <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  chrom_alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  gene_width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  chrom_color <span class="op">=</span> <span class="st">"#008888"</span>,</span>
<span>  gene_color <span class="op">=</span> <span class="st">"#0088ff"</span>,</span>
<span>  telomere_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_chrom_structure-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-chromosome-structures-and-gene-annotation">Plot chromosome structures and gene annotation<a class="anchor" aria-label="anchor" href="#plot-chromosome-structures-and-gene-annotation"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSMUSG00000042414"</span>, <span class="st">"ENSMUSG00000025935"</span>, <span class="st">"ENSMUSG00000048701"</span>, <span class="st">"ENSMUSG00000035385"</span><span class="op">)</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Prdm14"</span>, <span class="st">"Tram1"</span>, <span class="st">"Ccdc6"</span>, <span class="st">"Ccl2"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Vertical, annotate by name</span></span>
<span><span class="fu"><a href="reference/plot_chrom_genes.html">plot_chrom_genes</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  gene_table <span class="op">=</span> <span class="va">genes</span>,</span>
<span>  annotate <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"vertical"</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_chrom_genes-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-genomic-feature-density-heatmap">Plot genomic feature density heatmap<a class="anchor" aria-label="anchor" href="#plot-genomic-feature-density-heatmap"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Gene density heatmap</span></span>
<span><span class="fu"><a href="reference/plot_chrom_heatmap.html">plot_chrom_heatmap</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  feature <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"horizontal"</span>,</span>
<span>  palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#ffffff"</span>, <span class="st">"#0055aa"</span><span class="op">)</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.9</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_chrom_heatmap-1.png" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="id_6-deg-anno--viz">6. DEG Anno &amp; Viz<a class="anchor" aria-label="anchor" href="#id_6-deg-anno--viz"></a>
</h2>
<div class="section level3">
<h3 id="annotate-differentially-expressed-genes-degs-with-chromosome-positions">Annotate differentially expressed genes (DEGs) with chromosome positions<a class="anchor" aria-label="anchor" href="#annotate-differentially-expressed-genes-degs-with-chromosome-positions"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example DEGs GAnnoViz</span></span>
<span><span class="va">deg_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>,</span>
<span>  <span class="st">"example.deg"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="va">deg_file</span>,</span>
<span>  header <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>  na.strings <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">deg</span><span class="op">)</span></span>
<span><span class="co">#&gt;               GeneID baseMean log2FoldChange     lfcSE      stat   pvalue  padj</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000025198    23869      -6.000000 0.2843066 -21.10398 7.31e-74 1e-40</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000025272    40302       5.595767 0.2788325  20.06856 1.39e-64 1e-40</span></span>
<span><span class="co">#&gt; 3 ENSMUSG00000064264    13298      -5.534248 0.2955626 -18.72446 3.13e-53 1e-40</span></span>
<span><span class="co">#&gt; 4 ENSMUSG00000046432    55113       6.000000 0.3230247  18.57443 5.17e-52 1e-40</span></span>
<span><span class="co">#&gt; 5 ENSMUSG00000102989    28945       5.953063 0.3218916  18.49400 2.31e-51 1e-40</span></span>
<span><span class="co">#&gt; 6 ENSMUSG00000103364    13437       4.987586 0.2709416  18.40835 1.13e-50 1e-40</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Annotate DEGs with chromosome positions</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/anno_deg_chrom.html">anno_deg_chrom</a></span><span class="op">(</span></span>
<span>  deg_file <span class="op">=</span> <span class="va">deg_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  id_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  fc_col <span class="op">=</span> <span class="st">"log2FoldChange"</span>,</span>
<span>  use_strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  drop_unmapped <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;   chrom     start       end            gene_id     score strand</span></span>
<span><span class="co">#&gt; 1  chr3 116306719 116343630 ENSMUSG00000000340  1.456567      *</span></span>
<span><span class="co">#&gt; 2 chr17   6869070   6877078 ENSMUSG00000000579  2.696704      *</span></span>
<span><span class="co">#&gt; 3 chr11  54988941  55003855 ENSMUSG00000000594  1.488543      *</span></span>
<span><span class="co">#&gt; 4 chr10  77877781  77899456 ENSMUSG00000000730  1.592217      *</span></span>
<span><span class="co">#&gt; 5  chr8  71261093  71274068 ENSMUSG00000000791  1.344079      *</span></span>
<span><span class="co">#&gt; 6 chr11  83538670  83540181 ENSMUSG00000000982 -2.094984      *</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-differentially-expressed-genes-degs-volcano">Plot differentially expressed genes (DEGs) volcano<a class="anchor" aria-label="anchor" href="#plot-differentially-expressed-genes-degs-volcano"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Volcano plot</span></span>
<span><span class="fu"><a href="reference/plot_deg_volcano.html">plot_deg_volcano</a></span><span class="op">(</span></span>
<span>  deg_file <span class="op">=</span> <span class="va">deg_file</span>,</span>
<span>  id_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  fc_col <span class="op">=</span> <span class="st">"log2FoldChange"</span>,</span>
<span>  sig_col <span class="op">=</span> <span class="st">"padj"</span>,</span>
<span>  fc_threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sig_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  up_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  down_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  ns_color <span class="op">=</span> <span class="st">"#888888"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_deg_volcano-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-differentially-expressed-genes-degs-hyperhypo-distributions-by-chromosome">Plot differentially expressed genes (DEGs) hyper/hypo distributions by chromosome<a class="anchor" aria-label="anchor" href="#plot-differentially-expressed-genes-degs-hyperhypo-distributions-by-chromosome"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot chrom DEGs</span></span>
<span><span class="fu"><a href="reference/plot_deg_chrom.html">plot_deg_chrom</a></span><span class="op">(</span></span>
<span>  deg_file <span class="op">=</span> <span class="va">deg_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  id_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  fc_col <span class="op">=</span> <span class="st">"log2FoldChange"</span>,</span>
<span>  violin_scale <span class="op">=</span> <span class="st">"count"</span>,</span>
<span>  violin_border <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  point_shape <span class="op">=</span> <span class="fl">16</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  jitter_width <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff000088"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#00880088"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_deg_chrom-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-degs-updown-along-chromosomes">Plot DEGs up/down along chromosomes<a class="anchor" aria-label="anchor" href="#plot-degs-updown-along-chromosomes"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/plot_deg_exp.html">plot_deg_exp</a></span><span class="op">(</span></span>
<span>  deg_file <span class="op">=</span> <span class="va">deg_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  id_col <span class="op">=</span> <span class="st">"GeneID"</span>,</span>
<span>  fc_col <span class="op">=</span> <span class="st">"log2FoldChange"</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"horizontal"</span>,</span>
<span>  chrom_alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  chrom_color <span class="op">=</span> <span class="st">"#008888"</span>,</span>
<span>  bar_height <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  up_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  down_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  mark_style <span class="op">=</span> <span class="st">"point"</span>,</span>
<span>  line_width <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  line_height <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_deg_exp-1.png" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="id_7-snp-anno--plot">7. SNP Anno &amp; Plot<a class="anchor" aria-label="anchor" href="#id_7-snp-anno--plot"></a>
</h2>
<div class="section level3">
<h3 id="annotate-fst-slide-windows-with-genomic-features">Annotate FST slide windows with genomic features<a class="anchor" aria-label="anchor" href="#annotate-fst-slide-windows-with-genomic-features"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Annotate FST</span></span>
<span><span class="va">fst_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"example.fst"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="va">fst_file</span>,</span>
<span>  header <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>  na.strings <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span></span>
<span><span class="co">#&gt;   CHROM BIN_START BIN_END N_VARIANTS WEIGHTED_FST     MEAN_FST</span></span>
<span><span class="co">#&gt; 1  chr1         1   10000        119  0.034869746  0.027472144</span></span>
<span><span class="co">#&gt; 2  chr1     20001   30000        150 -0.009735326 -0.019103882</span></span>
<span><span class="co">#&gt; 3  chr1     40001   50000        587 -0.013269573 -0.004886184</span></span>
<span><span class="co">#&gt; 4  chr1     60001   70000        256  0.036700847  0.027643440</span></span>
<span><span class="co">#&gt; 5  chr1     80001   90000        842  0.020847846  0.011547497</span></span>
<span><span class="co">#&gt; 6  chr1    100001  110000        892  0.001750007  0.002121159</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/anno_fst_dmr.html">anno_fst_dmr</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  genomic_ranges <span class="op">=</span> <span class="va">fst_file</span>,</span>
<span>  chrom_col <span class="op">=</span> <span class="st">"CHROM"</span>,</span>
<span>  start_col <span class="op">=</span> <span class="st">"BIN_START"</span>,</span>
<span>  end_col <span class="op">=</span> <span class="st">"BIN_END"</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  ignore_strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"promoter"</span>, <span class="st">"UTR5"</span>, <span class="st">"gene"</span>, <span class="st">"exon"</span>, <span class="st">"intron"</span>, <span class="st">"CDS"</span>, <span class="st">"UTR3"</span>, <span class="st">"intergenic"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;   CHROM BIN_START BIN_END N_VARIANTS WEIGHTED_FST     MEAN_FST  anno_type</span></span>
<span><span class="co">#&gt; 1  chr1         1   10000        119  0.034869746  0.027472144 intergenic</span></span>
<span><span class="co">#&gt; 2  chr1     20001   30000        150 -0.009735326 -0.019103882 intergenic</span></span>
<span><span class="co">#&gt; 3  chr1     40001   50000        587 -0.013269573 -0.004886184 intergenic</span></span>
<span><span class="co">#&gt; 4  chr1     60001   70000        256  0.036700847  0.027643440 intergenic</span></span>
<span><span class="co">#&gt; 5  chr1     80001   90000        842  0.020847846  0.011547497 intergenic</span></span>
<span><span class="co">#&gt; 6  chr1    100001  110000        892  0.001750007  0.002121159 intergenic</span></span>
<span><span class="co">#&gt;   gene_id</span></span>
<span><span class="co">#&gt; 1        </span></span>
<span><span class="co">#&gt; 2        </span></span>
<span><span class="co">#&gt; 3        </span></span>
<span><span class="co">#&gt; 4        </span></span>
<span><span class="co">#&gt; 5        </span></span>
<span><span class="co">#&gt; 6</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotate-dmr-slide-windows-with-genomic-features">Annotate DMR slide windows with genomic features<a class="anchor" aria-label="anchor" href="#annotate-dmr-slide-windows-with-genomic-features"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Annotate DMR</span></span>
<span><span class="va">dmr_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"example.dmr"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dmr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  header <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>  na.strings <span class="op">=</span> <span class="st">"NA"</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dmr</span><span class="op">)</span></span>
<span><span class="co">#&gt;    chr   start     end strand pvalue   qvalue meth.diff</span></span>
<span><span class="co">#&gt; 1 chr1 4930001 4932000      *  1e-07 0.009117    89.850</span></span>
<span><span class="co">#&gt; 2 chr1 4936001 4938000      *  1e-08 0.005656   -82.314</span></span>
<span><span class="co">#&gt; 3 chr1 5670001 5672000      *  1e-09 0.011313   -69.908</span></span>
<span><span class="co">#&gt; 4 chr1 5914001 5916000      *  1e-09 0.001782   -93.310</span></span>
<span><span class="co">#&gt; 5 chr1 8576001 8578000      *  1e-08 0.009556   -68.589</span></span>
<span><span class="co">#&gt; 6 chr1 9098001 9100000      *  1e-07 0.009733   -68.756</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/anno_fst_dmr.html">anno_fst_dmr</a></span><span class="op">(</span></span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  genomic_ranges <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  chrom_col <span class="op">=</span> <span class="st">"chr"</span>,</span>
<span>  start_col <span class="op">=</span> <span class="st">"start"</span>,</span>
<span>  end_col <span class="op">=</span> <span class="st">"end"</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  ignore_strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"promoter"</span>, <span class="st">"UTR5"</span>, <span class="st">"gene"</span>, <span class="st">"exon"</span>, <span class="st">"intron"</span>, <span class="st">"CDS"</span>, <span class="st">"UTR3"</span>, <span class="st">"intergenic"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;    chr   start     end strand pvalue   qvalue meth.diff        anno_type</span></span>
<span><span class="co">#&gt; 1 chr1 4930001 4932000      *  1e-07 0.009117    89.850      gene,intron</span></span>
<span><span class="co">#&gt; 2 chr1 4936001 4938000      *  1e-08 0.005656   -82.314 gene,exon,intron</span></span>
<span><span class="co">#&gt; 3 chr1 5670001 5672000      *  1e-09 0.011313   -69.908      gene,intron</span></span>
<span><span class="co">#&gt; 4 chr1 5914001 5916000      *  1e-09 0.001782   -93.310       intergenic</span></span>
<span><span class="co">#&gt; 5 chr1 8576001 8578000      *  1e-08 0.009556   -68.589      gene,intron</span></span>
<span><span class="co">#&gt; 6 chr1 9098001 9100000      *  1e-07 0.009733   -68.756      gene,intron</span></span>
<span><span class="co">#&gt;                                       gene_id</span></span>
<span><span class="co">#&gt; 1 ENSMUSG00000033813(g),ENSMUSG00000104217(g)</span></span>
<span><span class="co">#&gt; 2 ENSMUSG00000033813(g),ENSMUSG00000104217(g)</span></span>
<span><span class="co">#&gt; 3                       ENSMUSG00000025905(g)</span></span>
<span><span class="co">#&gt; 4                                            </span></span>
<span><span class="co">#&gt; 5                       ENSMUSG00000025909(g)</span></span>
<span><span class="co">#&gt; 6                       ENSMUSG00000025909(g)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plot-snp-density-at-chromosome-level">Plot SNP density at chromosome level<a class="anchor" aria-label="anchor" href="#plot-snp-density-at-chromosome-level"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot SNP density</span></span>
<span><span class="fu"><a href="reference/plot_snp_density.html">plot_snp_density</a></span><span class="op">(</span></span>
<span>  fst_file <span class="op">=</span> <span class="va">fst_file</span>,</span>
<span>  LOG10 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>  density_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#0088ff"</span>, <span class="st">"#ff8800"</span>, <span class="st">"#ff0000"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_snp_density-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-genomic-weighted-fst-heatmap">Plot genomic weighted FST heatmap<a class="anchor" aria-label="anchor" href="#plot-genomic-weighted-fst-heatmap"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot weighted FST</span></span>
<span><span class="fu"><a href="reference/plot_snp_fst.html">plot_snp_fst</a></span><span class="op">(</span></span>
<span>  fst_file <span class="op">=</span> <span class="va">fst_file</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>  metric <span class="op">=</span> <span class="st">"fst_mean"</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"horizontal"</span>,</span>
<span>  palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#ffffff"</span>, <span class="st">"#aa00aa"</span><span class="op">)</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.9</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_snp_fst-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-genomic-fst-with-top-n-gene-annotations">Plot genomic FST with Top-N gene annotations<a class="anchor" aria-label="anchor" href="#plot-genomic-fst-with-top-n-gene-annotations"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Chromosome FST with Top-20 gene annotations on chr11</span></span>
<span><span class="fu"><a href="reference/plot_snp_anno.html">plot_snp_anno</a></span><span class="op">(</span></span>
<span>  fst_file <span class="op">=</span> <span class="va">fst_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  chrom_id <span class="op">=</span> <span class="st">"chr2"</span>,</span>
<span>  top_n <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  smooth_span <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  fst_color <span class="op">=</span> <span class="st">"#0088ff"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  connector_dx1 <span class="op">=</span> <span class="fl">2e4</span>,</span>
<span>  connector_dx2 <span class="op">=</span> <span class="fl">4e4</span>,</span>
<span>  gap_frac <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_snp_anno-1.png" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="id_8-dmg-anno--plot">8. DMG Anno &amp; Plot<a class="anchor" aria-label="anchor" href="#id_8-dmg-anno--plot"></a>
</h2>
<div class="section level3">
<h3 id="plot-differentially-methylated-regions-dmrs-hyperhypo-distributions-by-chromosome">Plot differentially methylated regions (DMRs) hyper/hypo distributions by chromosome<a class="anchor" aria-label="anchor" href="#plot-differentially-methylated-regions-dmrs-hyperhypo-distributions-by-chromosome"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot chrom DMRs</span></span>
<span><span class="fu"><a href="reference/plot_dmg_chrom.html">plot_dmg_chrom</a></span><span class="op">(</span></span>
<span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  violin_scale <span class="op">=</span> <span class="st">"count"</span>,</span>
<span>  violin_border <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  point_shape <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  jitter_width <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff880088"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#0088ff88"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_chrom_dmr-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-dmgs-hyperhypo-along-chromosomes">Plot DMGs hyper/hypo along chromosomes<a class="anchor" aria-label="anchor" href="#plot-dmgs-hyperhypo-along-chromosomes"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot DMG expression</span></span>
<span><span class="fu"><a href="reference/plot_dmg_exp.html">plot_dmg_exp</a></span><span class="op">(</span></span>
<span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  orientation <span class="op">=</span> <span class="st">"horizontal"</span>,</span>
<span>  chrom_alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  chrom_color <span class="op">=</span> <span class="st">"#008888"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  mark_style <span class="op">=</span> <span class="st">"line"</span>,</span>
<span>  line_width <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  line_height <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_dmg_exp-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-dmgs-circos-across-chromosomes">Plot DMGs circos across chromosomes<a class="anchor" aria-label="anchor" href="#plot-dmgs-circos-across-chromosomes"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot DMG circos</span></span>
<span><span class="fu"><a href="reference/plot_dmg_circos.html">plot_dmg_circos</a></span><span class="op">(</span></span>
<span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  label_type <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>  gene_table <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  y_transform <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  chrom_height <span class="op">=</span> <span class="fl">0.08</span>,</span>
<span>  chrom_color <span class="op">=</span> <span class="st">"#eeeeee"</span>,</span>
<span>  chrom_border <span class="op">=</span> <span class="st">"#cccccc"</span>,</span>
<span>  chrom_cex <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  gap_degree <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  x_tick_by <span class="op">=</span> <span class="fl">2e7</span>,</span>
<span>  axis_cex <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  last_gap_degree <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  scatter_height <span class="op">=</span> <span class="fl">0.15</span>,</span>
<span>  top_up <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  top_down <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  up_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  down_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  point_cex <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  annotation_height <span class="op">=</span> <span class="fl">0.18</span>,</span>
<span>  label_cex <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  label_rotate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  label_font <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  connector_lwd <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  connector_col <span class="op">=</span> <span class="st">"#333333"</span>,</span>
<span>  connector_len <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  connector_elbow <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_dmg_circos-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-dmgs-manhattan-plot-across-chromosomes">Plot DMGs manhattan plot across chromosomes<a class="anchor" aria-label="anchor" href="#plot-dmgs-manhattan-plot-across-chromosomes"></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot DMG Manhattan</span></span>
<span><span class="fu"><a href="reference/plot_dmg_manhattan.html">plot_dmg_manhattan</a></span><span class="op">(</span></span>
<span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  gene_table <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_type <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>  label_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  y_transform <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  chromosome_spacing <span class="op">=</span> <span class="fl">1e6</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  label_top_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  gap_frac <span class="op">=</span> <span class="fl">0.04</span>,</span>
<span>  connector_dx1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  connector_dx2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  connector_elbow <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  connector_tilt_frac <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_dmg_manhattan-1.png" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="plot-chromosomal-dmgs-trend">Plot chromosomal DMGs trend<a class="anchor" aria-label="anchor" href="#plot-chromosomal-dmgs-trend"></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot DMR trend</span></span>
<span><span class="fu"><a href="reference/plot_dmg_trend.html">plot_dmg_trend</a></span><span class="op">(</span></span>
<span>  chrom_id <span class="op">=</span> <span class="st">"chr1"</span>,</span>
<span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span>
<span>  smooth_span <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plot_dmg_trend-1.png" style="display: block; margin: auto;"></p>
</div>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/benben-miao/GAnnoViz/">Browse source code</a></li>
<li><a href="https://github.com/benben-miao/GAnnoViz/issues">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GAnnoViz</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Benben Miao <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Benben Miao.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
