<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot DMGs manhattan plot across chromosomes — plot_dmg_manhattan • GAnnoViz</title><!-- katex math --><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.css" integrity="sha384-nB0miv6/jRmo5UMMR1wu3Gz6NLsoTkbqJghGIsx//Rlm+ZU03BU6SQNC66uf4l5+" crossorigin="anonymous"><script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/katex.min.js" integrity="sha384-7zkQWkzuo3B5mTepMUcHkMB5jZaolc2xDwL6VFqjFALcbeS9Ggm/Yr2r3Dy4lfFg" crossorigin="anonymous"></script><script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.11/dist/contrib/auto-render.min.js" integrity="sha384-43gviWU0YVjaDtb/GhzOouOXtZMP/7XUzwPTstBeZFe/+rCMvRwr4yROQP43s0Xk" crossorigin="anonymous" onload="renderMathInElement(document.body);"></script><script src="../katex-auto.js"></script><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/JetBrains_Mono-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Plot DMGs manhattan plot across chromosomes — plot_dmg_manhattan"><meta name="description" content="Plot genome-wide DMGs manhattan."><meta property="og:description" content="Plot genome-wide DMGs manhattan."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GAnnoViz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-passport"></span> Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-file-code"></span> Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Tutorial.html"><span class="fa fa-file-code"></span> Tutorial</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/benben-miao/GAnnoViz/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plot DMGs manhattan plot across chromosomes</h1>
      <small class="dont-index">Source: <a href="https://github.com/benben-miao/GAnnoViz/blob/HEAD/R/plot_dmg_manhattan.R"><code>R/plot_dmg_manhattan.R</code></a></small>
      <div class="d-none name"><code>plot_dmg_manhattan.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plot genome-wide DMGs <b><em>manhattan</em></b>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_dmg_manhattan</span><span class="op">(</span></span>
<span>  <span class="va">dmr_file</span>,</span>
<span>  <span class="va">gff_file</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  gene_table <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_type <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>  label_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  y_transform <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  chromosome_spacing <span class="op">=</span> <span class="fl">1e+06</span>,</span>
<span>  hyper_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  hypo_color <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  label_top_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  gap_frac <span class="op">=</span> <span class="fl">0.04</span>,</span>
<span>  connector_dx1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  connector_dx2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  connector_elbow <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  connector_tilt_frac <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-dmr-file">dmr_file<a class="anchor" aria-label="anchor" href="#arg-dmr-file"></a></dt>
<dd><p>DEG table from <b><em>MethylKit</em></b> analysis.</p></dd>


<dt id="arg-gff-file">gff_file<a class="anchor" aria-label="anchor" href="#arg-gff-file"></a></dt>
<dd><p>Genomic structural annotation <b><code class="file">GFF3/GTF</code></b> file path. If provided and <code>label_col</code> is <code>NULL</code>, will try to label by overlapped/nearest <code>gene_id</code>.</p></dd>


<dt id="arg-format">format<a class="anchor" aria-label="anchor" href="#arg-format"></a></dt>
<dd><p>Format of GFF3/GTF file. (<b><em>"auto"</em></b>, "gff3", "gtf").</p></dd>


<dt id="arg-gene-table">gene_table<a class="anchor" aria-label="anchor" href="#arg-gene-table"></a></dt>
<dd><p>Optional gene ID/name mapping table (first two columns: id, name). If provided, will use gene name for labels when possible.</p></dd>


<dt id="arg-label-type">label_type<a class="anchor" aria-label="anchor" href="#arg-label-type"></a></dt>
<dd><p>Label by gene name or gene id. (<b><em>"name"</em></b>, "id"). If <code>"name"</code> but no gene names can be inferred from <code>gff_file</code>, please provide <code>gene_table</code>.</p></dd>


<dt id="arg-label-col">label_col<a class="anchor" aria-label="anchor" href="#arg-label-col"></a></dt>
<dd><p>Column name used for annotation labels. If <code>NULL</code>, uses <code>chr:start-end</code>.</p></dd>


<dt id="arg-y-transform">y_transform<a class="anchor" aria-label="anchor" href="#arg-y-transform"></a></dt>
<dd><p>Y-axis transformation for <code>meth.diff</code>. (<b><em>"none"</em></b>, "log2", "log10").</p></dd>


<dt id="arg-chromosome-spacing">chromosome_spacing<a class="anchor" aria-label="anchor" href="#arg-chromosome-spacing"></a></dt>
<dd><p>Gap width (bp) inserted between chromosomes on x-axis. (<b><em>1e6</em></b>).</p></dd>


<dt id="arg-hyper-color">hyper_color<a class="anchor" aria-label="anchor" href="#arg-hyper-color"></a></dt>
<dd><p>Color for hyper-methylated points. (<b><em>"#ff0000"</em></b>).</p></dd>


<dt id="arg-hypo-color">hypo_color<a class="anchor" aria-label="anchor" href="#arg-hypo-color"></a></dt>
<dd><p>Color for hypo-methylated points. (<b><em>"#008800"</em></b>).</p></dd>


<dt id="arg-point-size">point_size<a class="anchor" aria-label="anchor" href="#arg-point-size"></a></dt>
<dd><p>Point size. (<b><em>1</em></b>).</p></dd>


<dt id="arg-point-alpha">point_alpha<a class="anchor" aria-label="anchor" href="#arg-point-alpha"></a></dt>
<dd><p>Point alpha. (<b><em>0.5</em></b>).</p></dd>


<dt id="arg-label-top-n">label_top_n<a class="anchor" aria-label="anchor" href="#arg-label-top-n"></a></dt>
<dd><p>Number of top positive and top negative DMGs to label. (<b><em>10</em></b>).</p></dd>


<dt id="arg-label-size">label_size<a class="anchor" aria-label="anchor" href="#arg-label-size"></a></dt>
<dd><p>Text size for labels. (<b><em>3</em></b>).</p></dd>


<dt id="arg-gap-frac">gap_frac<a class="anchor" aria-label="anchor" href="#arg-gap-frac"></a></dt>
<dd><p>Minimum vertical gap between labels (fraction of y-range). (<b><em>0.04</em></b>).</p></dd>


<dt id="arg-connector-dx-">connector_dx1<a class="anchor" aria-label="anchor" href="#arg-connector-dx-"></a></dt>
<dd><p>First connector horizontal offset (bp). Default adapts to genome size.</p></dd>


<dt id="arg-connector-dx-">connector_dx2<a class="anchor" aria-label="anchor" href="#arg-connector-dx-"></a></dt>
<dd><p>Second connector horizontal offset (bp). Default adapts to genome size.</p></dd>


<dt id="arg-connector-elbow">connector_elbow<a class="anchor" aria-label="anchor" href="#arg-connector-elbow"></a></dt>
<dd><p>Scale factor applied to <code>connector_dx2</code>. (<b><em>0.8</em></b>).</p></dd>


<dt id="arg-connector-tilt-frac">connector_tilt_frac<a class="anchor" aria-label="anchor" href="#arg-connector-tilt-frac"></a></dt>
<dd><p>Tilt amplitude for the second connector segment as a fraction of <code>gap_frac</code>. (<b><em>0.2</em></b>).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <b><em>ggplot</em></b> object of DMG manhattan plot.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>benben-miao</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">dmr_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>  <span class="st">"example.dmr"</span>,</span></span>
<span class="r-in"><span>  package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gff_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>  <span class="st">"example.gff3.gz"</span>,</span></span>
<span class="r-in"><span>  package <span class="op">=</span> <span class="st">"GAnnoViz"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">plot_dmg_manhattan</span><span class="op">(</span></span></span>
<span class="r-in"><span>  dmr_file <span class="op">=</span> <span class="va">dmr_file</span>,</span></span>
<span class="r-in"><span>  gff_file <span class="op">=</span> <span class="va">gff_file</span>,</span></span>
<span class="r-in"><span>  format <span class="op">=</span> <span class="st">"auto"</span>,</span></span>
<span class="r-in"><span>  gene_table <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  label_type <span class="op">=</span> <span class="st">"name"</span>,</span></span>
<span class="r-in"><span>  label_col <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  y_transform <span class="op">=</span> <span class="st">"none"</span>,</span></span>
<span class="r-in"><span>  chromosome_spacing <span class="op">=</span> <span class="fl">1e6</span>,</span></span>
<span class="r-in"><span>  hyper_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span></span>
<span class="r-in"><span>  hypo_color <span class="op">=</span> <span class="st">"#008800"</span>,</span></span>
<span class="r-in"><span>  point_size <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span></span>
<span class="r-in"><span>  label_top_n <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>  gap_frac <span class="op">=</span> <span class="fl">0.04</span>,</span></span>
<span class="r-in"><span>  connector_dx1 <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  connector_dx2 <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  connector_elbow <span class="op">=</span> <span class="fl">0.8</span>,</span></span>
<span class="r-in"><span>  connector_tilt_frac <span class="op">=</span> <span class="fl">0.2</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Import genomic features from the file as a GRanges object ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prepare the 'metadata' data frame ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Make the TxDb object ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-plt img"><img src="plot_dmg_manhattan-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Benben Miao.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

